src/status-model.ts

```diff
@@ -69,9 +68,8 @@ export class Status {
     this.run_link_title = `${this.workflow} / ${this.job} ${differentiator}`
     this.run_link_url = `${process.env.GITHUB_SERVER_URL}/${owner}/${repo}/actions/runs/${process.env.GITHUB_RUN_ID}/attempts/${process.env.GITHUB_RUN_ATTEMPT}`
 
-    const event = readEventDetails()
-    this.sha = event.sha
-    this.issue_number = event.issue_number
+    this.sha = github.context.payload.pull_request?.head?.sha
+    this.issue_number = github.context.payload.pull_request?.number as number
```

Comment:
Question:

>`.pull_request?.number as number`

is using the *optional chaining* syntax (kinda like in Kotlin), and then a *type assertion*, asserting that the value of `number` is of type `number`. What happens if the optional chain fails and the result ends up being `undefined`? How does that interact with the type assertion?

Reply:
`as number` is a cast rather than assertion, but this is lazy coding and ignoring the possibility of `undefined`.

It probably ought to be something like this to default `undefined` values to `0`:
```suggestion
    this.issue_number = github.context.payload.pull_request?.number || 0
```